source/manuals/programming-languages/python/python.html.md.erb

```diff
@@ -0,0 +1,94 @@
+---
+title: Writing Python at GDS
+expires: living
+---
+
+# <%= current_page.data.title %>
+
+
+### About this manual:
+
+This manual is designed to aid developers in writing Python code that is clear and consistent, within, and across,
+projects at GDS.
+
+The [Google Python style guide][GPSG] outlines sensible conventions for
+writing clear and concise Python code and we would suggest using this as a reference unless something is explicitly
+mentioned in the [GDS Python Style Guide][GDSPSG].
+
+
+### IDEs
+
+[JetBrains PyCharm][PyCharm] is commonly used at GDS but IDE choice is not enforced as part of this guide.
+
+Whatever your choice something with decent Python support is going to make complying with the suggestions in this guide
+far easier. If you haven't worked with Python before then your first port of call should be the people you're working
+with. This will enable you to get help quickly when you need it.
+
+
+### Consistency
+
+Mixing different styles and code patterns reduces confidence in a code base. If yours is the code that looks out of
+place, no matter how 'good' a piece of code it is, other developers will wonder why it's out of line with other parts of
+the code base.
+
+If you really want to change a code pattern then you should be prepared to change all of the code in a refactor pull
+request.
+
+_Please do not introduce new patterns without also updating existing code._
+
+_Please consider new styles with your team before taking the time to write them out. You may be disappointed._
+
+With regards to the [GDS Python Style Guide][GDSPSG], as with the _vast_ majority of programming language style guides out
+there, consistency with existing styles and paradigms is valued over conformity to the guide.
+
+
+### Python 3
+
+Prefer Python 3. Python 2 will [reach end of life in 2020][PEP373].
+Python 3 is now well-supported by application frameworks and libraries, and is commonly used in production.
+
+If you've never migrated a project from Python2 to Python3 there are plenty of resources online and it is usually
+surprisingly easy.
+
+
+### PEP8
+
+The only allowable exceptions to [PEP8][PEP8] should be detailed in the [GDS Python Style Guide][GDSPSG]or in the
```

Comment:
Missing space after [GDS Python Style Guide][GDSPSG]

Reply:
That quote is directly lifted from pep8.
https://www.python.org/dev/peps/pep-0008/#a-foolish-consistency-is-the-hobgoblin-of-little-minds
So this is more to imply that direct exceptions to pep8 _must_ be documented (like line length 120 opposing pep8s <=79)

Could change the wording to something like:

> Specific exceptions to PEP8 must be documented in the ...
?

Reply:
Done the missing space